samples/extensions/fragment_shading_rate_dynamic/fragment_shading_rate_dynamic.cpp

```diff
@@ -0,0 +1,909 @@
+	vertex_input_state.pVertexBindingDescriptions           = vertex_input_bindings.data();
+	vertex_input_state.vertexAttributeDescriptionCount      = static_cast<uint32_t>(vertex_input_attributes.size());
+	vertex_input_state.pVertexAttributeDescriptions         = vertex_input_attributes.data();
+
+	pipeline_create_info.pVertexInputState = &vertex_input_state;
+
+	pipeline_create_info.layout     = pipeline_layout;
+	pipeline_create_info.renderPass = render_pass;
+
+	// Sky-sphere
+	shader_stages[0] = load_shader("fragment_shading_rate_dynamic/scene.vert", VK_SHADER_STAGE_VERTEX_BIT);
+	shader_stages[1] = load_shader("fragment_shading_rate_dynamic/scene.frag", VK_SHADER_STAGE_FRAGMENT_BIT);
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &pipeline_create_info, nullptr, &pipelines.skysphere));
+
+	// Objects
+	// Enable depth test and write
+	depth_stencil_state.depthWriteEnable = VK_TRUE;
+	depth_stencil_state.depthTestEnable  = VK_TRUE;
+	// Flip cull mode
+	rasterization_state.cullMode = VK_CULL_MODE_FRONT_BIT;
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &pipeline_create_info, nullptr, &pipelines.sphere));
+}
+
+void FragmentShadingRateDynamic::prepare_uniform_buffers()
+{
+	uniform_buffers.scene = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                            sizeof(ubo_scene),
+	                                                            VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT,
+	                                                            VMA_MEMORY_USAGE_CPU_TO_GPU);
+	update_uniform_buffers();
+}
+
+void FragmentShadingRateDynamic::update_uniform_buffers()
+{
+	ubo_scene.projection          = camera.matrices.perspective;
+	ubo_scene.modelview           = camera.matrices.view * glm::mat4(1.0f);
+	ubo_scene.skysphere_modelview = camera.matrices.view;
+	ubo_scene.color_shading_rate  = color_shading_rate;
+	uniform_buffers.scene->convert_and_update(ubo_scene);
+}
+
+void FragmentShadingRateDynamic::draw()
+{
+	VkSemaphore        semaphore{VK_NULL_HANDLE};
+	const VkSemaphore *old_semaphore{VK_NULL_HANDLE};
+	auto               semaphore_create = vkb::initializers::semaphore_create_info();
+	vkCreateSemaphore(device->get_handle(), &semaphore_create, VK_NULL_HANDLE, &semaphore);
+
+	ApiVulkanSample::prepare_frame();
+	assert(submit_info.signalSemaphoreCount == 1);
+	old_semaphore = submit_info.pSignalSemaphores;
+	std::vector<VkSemaphore> semaphores{submit_info.pSignalSemaphores[0], semaphore};
+
+	submit_info.commandBufferCount   = 1;
+	submit_info.pCommandBuffers      = &draw_cmd_buffers[current_buffer];
+	submit_info.signalSemaphoreCount = 2;
+	submit_info.pSignalSemaphores    = semaphores.data();
+
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, VK_NULL_HANDLE));
+	ApiVulkanSample::submit_frame();
+
+	const VkPipelineStageFlags wait_mask           = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
```

Comment:
`VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT` -- we want to block COMPUTE until the semaphore signals.